Src/FluentAssertions/Execution/AssertionChain.cs

```diff
@@ -0,0 +1,355 @@
+﻿using System;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using FluentAssertions.Common;
+
+namespace FluentAssertions.Execution;
+
+public sealed class AssertionChain
```

Comment:
Would be great to have some more API documentation on this new core class.
There are also some public members left without documentation.

Reply:
I documented the class itself, but copied most of the other members directly from the previous version of `AssertionScope`.